src/lib/Libpython/pbs_python_svr_internal.c

```diff
@@ -328,12 +328,31 @@ typedef struct server_vnodes_t {
 } server_vnodes_t;
 static	server_vnodes_t server_vnodes;
 
+/**
+ * @brief
+ * 	This is a function that logs the contents of the list
+ *	headed by 'phead'.
+ *
+ * @param[in] head_str - some string that gets printed spearheading the list.
+ * @param[in] phead - header of the list to be printed.
+ *
+ * @return	void
+ *
+ */
 void
 print_svrattrl_list(char *head_str, pbs_list_head *phead)
 {
 	svrattrl *plist = NULL;
 	int i;
 
+	if ((head_str == NULL) || (phead == NULL) || (log_event_mask == NULL)) {
+		return;
+	}
+
+	if ((*log_event_mask & PBSEVENT_DEBUG3) == 0) {
```

Comment:
I really wanted to do this so as to eliminate unnecessary looping if nothing is going to get logged anyway. I can just imagine what if a job request say 100K nodes, so that will require looping 100K times unnecessarily.  It can still potentially timeout the hook. Another thing is that since an execjob_begin and execjob_end hook can block the mom while waiting to finish (since it is executed by main mom), I'd like to make this type of hook not have any additional delay.